tutorials/btp-app-events-app-setup-s4hc/btp-app-events-app-setup-s4hc.md

```diff
@@ -0,0 +1,378 @@
+            "kind": "odata-v2",
+            "model": "srv/external/API_BUSINESS_PARTNER-extended",
+            ...
+          },
+          "messaging": {
+            "kind": "local-messaging",
+            "[production]": {
+                "kind": "enterprise-messaging-shared",
+                "format": "cloudevents",
+                "publishPrefix": "$namespace/ce/",
+                "subscribePrefix": "+/+/+/ce/"
+            }
+          }
+        },
+        ...
+    ```
+
+    What happens here?
+
+    * `"model": "srv/external/API_BUSINESS_PARTNER-extended"` specifies the model that is extended with event declarations for business partners with the respective SAP S4/HANA topic annotations.
+    * `"kind": "local-messaging"` Allows services running in the same process to send and receive messages locally. Other options for local testing are [`file` for services running in different processes](https://cap.cloud.sap/docs/guides/messaging/#1-use-file-based-messaging-in-development) and [connecting to an event mesh instance on BTP](https://cap.cloud.sap/docs/guides/messaging/event-mesh#run-tests-in-hybrid-setup).
+    * In production we will use the [SAP Event Mesh service](https://help.sap.com/docs/SAP_EM?locale=en-US), hence the line `"kind": "enterprise-messaging-shared"`.
+    * `"format": "cloudevents"` configures the format of the event data, which allows [CAP to fill in required fields](https://cap.cloud.sap/docs/guides/messaging/#cloudevents) behind the scenes.
+    * The `publishPrefix` is specified for all events emitted from the application (`$namespace/ce/`). This is not used in this application as we don't emit any events.
+    * `subscribePrefix` is specified to match all the Cloud Events sent to the subaccount. The `ce/` part in the expression `+/+/+/ce/` refers to the cloud events.
+
+    > Namespace info for `subscribePrefix` and `publishPrefix`
+
+    >   The `$namespace` of the event topic name will be generated for SAP S/4HANA when deploying the `cpapp-bupa-messaging` service where we specify the client name, for example, `sap/S4HANAOD/RISK/`. To match the namespace we are specifying the pattern with 3 `+`'s separated by `/`.
+
+3. Since SAP S/4HANA integrates SAP Event Mesh for messaging, we need an AMQP (Advanced Message Queuing Protocol) client so our app can communicate with SAP Event Mesh. Run the following command to install it:
+
+    ```Shell/Bash
+    npm install @sap/xb-msg-amqp-v100
+    ```
+
+[VALIDATE_1]
+[ACCORDION-END]
+---
+[ACCORDION-BEGIN [Step 2: ](Add a status field)]
+Currently, business partner (that is, supplier) data is stored only in the SAP S/4HANA Cloud system and it would be slow to call the Business Partner API every time we want to show the status. To keep it fast, we will add a `status` field to reflect the assessment status of the supplier. Let's update the application logic with event handling to keep this value updated as it changes in the remote SAP S/4HANA Cloud system.
```

Comment:
Do not say "we will add"  Just say "add"

Reply:
done